src/Illuminate/Database/Query/Builder.php

```diff
@@ -1784,6 +1784,10 @@ public function orHavingRaw($sql, array $bindings = [])
      */
     public function orderBy($column, $direction = 'asc')
     {
+        if (! in_array(strtolower($direction), ['asc', 'desc'])) {
```

Comment:
You should make a string lowercase before check:

```php
$direction = strtolower($direction);
```

and change the following line:

```diff
- 'direction' => strtolower($direction) === 'asc' ? 'asc' : 'desc',
+ 'direction' => $direction === 'asc' ? 'asc' : 'desc',
```

Also, I think this PR should be sent to the master branch (with tests), because it's a breaking change (Laravel will no longer use `desc` direction when you pass invalid parameter).

Reply:
@bonzai 

1.Missing PHPDoc -> done
2.You should make a string lowercase before check: -> done
3.ok, I will move PR to 5.8
4.I am not sure if tests are required because other "InvalidArgumentException" from "src/Illuminate/Database/Query/Builder.php" do not have tests.